homeassistant/components/insteon_plm.py

```diff
@@ -6,115 +6,176 @@
         #
-        if isinstance(device['platform'], list):
-            plm.protocol.devices.add_override(
-                device['address'], 'capabilities', device['platform'])
-        else:
-            plm.protocol.devices.add_override(
-                device['address'], 'capabilities', [device['platform']])
+        if device_override.get('cat', False):
+            plm.devices.add_override(device_override['address'],
+                                     'cat',
+                                     device_override['cat'])
+        if device_override.get('subcat', False):
+            plm.devices.add_override(device_override['address'],
+                                     'subcat',
+                                     device_override['subcat'])
+        if device_override.get('firmware', False):
+            plm.devices.add_override(device_override['address'],
+                                     'product_key',
+                                     device_override['firmware'])
+        if device_override.get('product_key', False):
+            plm.devices.add_override(device_override['address'],
+                                     'product_key',
+                                     device_override['product_key'])
 
     hass.data['insteon_plm'] = plm
 
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, plm.close)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, conn.close)
 
-    plm.protocol.devices.add_device_callback(async_plm_new_device, {})
+    plm.devices.add_device_callback(async_plm_new_device)
 
     return True
 
 
-def common_attributes(entity):
+def common_attributes(entity, state):
     """Return the device state attributes."""
-    attributes = {}
-    attributekeys = {
-        'address': 'INSTEON Address',
-        'description': 'Description',
-        'model': 'Model',
-        'cat': 'Category',
-        'subcat': 'Subcategory',
-        'firmware': 'Firmware',
-        'product_key': 'Product Key'
+    attributes = {
+        'INSTEON Address': entity.address.human,
+        'Description': entity.description,
+        'Model': entity.model,
+        'Category': '{:02x}'.format(entity.cat),
+        'Subcategory': '{:02x}'.format(entity.subcat),
+        'Product Key / Firmware': '{:02x}'.format(entity.product_key),
+        'Group': '{:02x}'.format(state.group)
     }
+    return attributes
 
-    hexkeys = ['cat', 'subcat', 'firmware']
 
-    for key in attributekeys:
-        name = attributekeys[key]
-        val = entity.get_attr(key)
-        if val is not None:
-            if key in hexkeys:
-                attributes[name] = hex(int(val))
-            else:
-                attributes[name] = val
-    return attributes
+State = collections.namedtuple('Product', 'stateType platform')
+
+
+class IPDB(object):
+    """Embodies the INSTEON Product Database static data
+and access methods."""
+
+    from insteonplm.states.onOff import (OnOffSwitch,
```

Comment:
This is not allowed. Needs to happen inside functions.

Reply:
Per you comment above regarding each PR to 1 feature, the issue here is that the underlying library was refactored because it could only support very simple devices. In order to support that change, significant changes had to be performed to this code. 

Done. Put inside __init__ to remove class attribute anyway.